clidocstool_md_test.go

```diff
@@ -39,15 +40,38 @@ func TestGenMarkdownTree(t *testing.T) {
 	require.NoError(t, err)
 	require.NoError(t, c.GenMarkdownTree(buildxCmd))
 
-	for _, tt := range []string{"buildx.md", "buildx_build.md", "buildx_stop.md"} {
-		tt := tt
-		t.Run(tt, func(t *testing.T) {
-			bres, err := os.ReadFile(filepath.Join(tmpdir, tt))
+	seen := make(map[string]struct{})
+
+	require.NoError(t, filepath.Walk("fixtures", func(path string, info fs.FileInfo, err error) error {
+		fname := filepath.Base(path)
+		// ignore dirs, .pre.md files and any file that is not a .md file
+		if info.IsDir() || !strings.HasSuffix(fname, ".md") || strings.HasSuffix(fname, ".pre.md") {
+			return nil
+		}
+		seen[fname] = struct{}{}
```

Comment:
Perhaps slightly better if this was put inside the `t.Run()` - technically that would guarantee that the test has run (although in practice it doesn't make a real difference of course)